demo/app/src/App.js

```diff
@@ -0,0 +1,349 @@
+  const [offset, setOffset] = useState(defaults.offset);
+  // plugins
+  const [autoRefreshPlugin, setAutoRefreshPlugin] = useState(false);
+  const [autoRenderPlugin, setAutoRenderPlugin] = useState(false);
+  const [debugPlugin, setDebugPlugin] = useState(false);
+  const [loggingPlugin, setLoggingPlugin] = useState(false);
+  const [responsivePlugin, setResponsivePlugin] = useState(false);
+  const [stickyPlugin, setStickyPlugin] = useState(false);
+  // ads
+  const [selectedDesktopSizes, setSelectedDesktopSizes] = useState(
+    defaults.sizes.desktop
+  );
+  const [selectedTabletSizes, setSelectedTabletSizes] = useState(
+    defaults.sizes.tablet
+  );
+  const [selectedMobileSizes, setSelectedMobileSizes] = useState(
+    defaults.sizes.mobile
+  );
+
+  const selectedAds = ads =>
+    ads
+      .filter(ad => ad.checked === true)
+      .map(ad => [ad.w, ad.h])
+      .reduce((ads, ad) => ads.concat([ad]), []);
+
+  const buildSetSelection = (setSelected, sizes) => {
+    return event => {
+      const newSelectedSizes = [...sizes];
+      const [currentX, currentY] = event.target.value.split("x");
+
+      newSelectedSizes.forEach(ad => {
+        if ((ad.w === parseInt(currentX)) & (ad.h === parseInt(currentY))) {
+          return (ad.checked = !ad.checked);
+        }
+      });
+      setSelected(newSelectedSizes);
+    };
+  };
+
+  const setSelectionDesktop = buildSetSelection(
+    setSelectedDesktopSizes,
+    selectedDesktopSizes
+  );
+  const setSelectionTablet = buildSetSelection(
+    setSelectedTabletSizes,
+    selectedTabletSizes
+  );
+  const setSelectionMobile = buildSetSelection(
+    setSelectedMobileSizes,
+    selectedMobileSizes
+  );
+
+  const getAdConfiguration = () => {
+    const desktopAds = selectedAds(selectedDesktopSizes);
+    const tabletAds = selectedAds(selectedTabletSizes);
+    const mobileAds = selectedAds(selectedMobileSizes);
+
+    return {
+      path: path,
+      sizes: {
+        mobile: mobileAds,
+        tablet: tabletAds,
+        desktop: desktopAds
+      },
+      targeting: JSON.parse(targeting)
+    };
+  };
+
+  const getBucketConfiguration = () => {
+    let plugins = [];
+
+    if (loggingPlugin) plugins.push("Logging");
+    if (autoRenderPlugin) plugins.push("AutoRender");
+    if (autoRefreshPlugin) plugins.push("AutoRefresh");
+    if (debugPlugin) plugins.push("Debug");
+    if (stickyPlugin) plugins.push("Sticky");
+
+    return {
+      plugins: plugins.map(plugin => window.AdJS.Plugins[plugin]),
+      defaults: {
+      breakpoints: {
+          mobile: { from: 0, to: 350 },
+          tablet: { from: 351, to: 780 },
+          desktop: { from: 781, to: Infinity }
+        },
+        refreshRateInSeconds: refreshRate,
+        offset: offset,
+        targeting: JSON.parse(targeting)
+      }
+    };
+  };
+
+  const setSourceCodePreview = returnCode => {
+    let sourceCode = `
+const mainPage = new AdJS.Bucket(AdJS.Networks.${provider}, ${JSON.stringify(
```

Comment:
Couple alignment issues with template strings

Reply:
This is updated to contain more source code, starting with no indent, the goal is to have the whole source code preview nicely left align.